server/datastore/mysql/hosts.go

```diff
@@ -651,6 +651,13 @@ func (ds *Datastore) EnrollHost(ctx context.Context, osqueryHostID, nodeKey stri
 				return backoff.Permanent(ctxerr.Errorf(ctx, "host identified by %s enrolling too often", osqueryHostID))
 			}
 			hostID = int64(host.ID)
+
+			if host.TeamID != teamID {
```

Comment:
`host.TeamID` and `teamID` are pointers.

Reply:
And while we are at it, let's add a test that re-enrolling to the same team won't clear the policy membership.

Reply:
🤦 thanks Lucas. Fix + test added @ 2def2dd785b706241fda59e31a36256d345046d8